File: Data Structures & Algorithms/binary-tree-postorder-traversal/submission-0.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func postorderTraversal(root *TreeNode) []int {
	result := []int{}

	var aux func(node *TreeNode)
	aux = func(node *TreeNode) {
		if node == nil {
			return
		}
		aux(node.Left)
		aux(node.Right)
		result = append(result, node.Val)
	}

	aux(root)
	return result
}

File: Data Structures & Algorithms/binary-tree-inorder-traversal/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> result = new ArrayList<>();
        return aux(root, result);
    }

    private List<Integer> aux(TreeNode root, List<Integer> result) {
        if (root == null) return result;

        aux(root.left, result);
        result.add(root.val);
        aux(root.right, result);
        return result;
    }
}